GVFS/GVFS.Common/Tracing/ITracer.cs

```diff
@@ -9,6 +9,8 @@ public interface ITracer : IDisposable
         ITracer StartActivity(string activityName, EventLevel level, EventMetadata metadata);
         ITracer StartActivity(string activityName, EventLevel level, Keywords startStopKeywords, EventMetadata metadata);
 
+        void SetGitCommandSessionId(string sessionId);

```

Comment:
Adding this here seems like a hack to me but when the lock request comes in we only have the `ITracer` object.  We could cast to a `JsonTracer` so we don't have to add this here but that to me would also be a hack.  There didn't seem to be any clean way to get the `GitCommandSessionId` to the `TelemetryDaemonEventListener` so if anyone has some brilliant ideas please share.